feature_engine/encoding/one_hot.py

```diff
@@ -210,7 +215,12 @@ def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
 
         X = check_X(X)
         variables_ = self._check_or_select_variables(X)
-        _check_contains_na(X, variables_)
+        
+        # check if dataset contains na if missing_value is 'raise' or empty
+        if self.missing_values == "raise":
+            _check_contains_na(X, variables_)
+        elif self.missing_values == "encode":
+            X = X.fillna('novalue_temp')
```

Comment:
This is looking really good :)

Could we use a different string? like"nan" or "missing_value"?

We should probable only fillna in `X[variables_]`